src/Agent.Worker/TestResults/TestResultsServer.cs

```diff
@@ -88,5 +98,24 @@ public void InitializeServer(VssConnection connection)
         {
             return await TestHttpClient.CreateTestSubResultAttachmentAsync(reqModel, projectName, testRunId, testCaseResultId, testSubResultId, cancellationToken);
         }
+
+        private const string EnablePublishToTcmServiceDirectlyFromTaskFF = "TestManagement.Server.EnablePublishToTcmServiceDirectlyFromTask";
+
+        private static bool GetFeatureFlagState(FeatureAvailabilityHttpClient featureAvailabilityHttpClient, string FFName)
+        {
+            try
+            {
+                var featureFlag = featureAvailabilityHttpClient?.GetFeatureFlagByNameAsync(FFName).Result;
+                if (featureFlag != null && featureFlag.EffectiveState.Equals("Off", StringComparison.OrdinalIgnoreCase))
```

Comment:
simplify logic. and there's an issue if featureflag is not defined in that case you are returning true.

if it's not null and EffectState On -> return true

otherwise return false.

Reply:
yes, we should keep default FF value as false and true only if FF is ON.

Reply:
Changed it.